EfYou/EntityServices/EntityServiceOfT.cs

```diff
@@ -57,6 +61,26 @@ public virtual List<T> Get(List<long> ids, List<string> includes, List<OrderBy>
             return Get(ids, includes, orderBys, null);
         }
 
+        public T GetLast(List<long> ids)
+        {
+            return GetLast(ids, null, null);
+        }
+
+        public T GetLast(List<long> ids, List<string> includes)
+        {
+            return GetLast(ids, includes, null);
+        }
+
+        public T GetLast(List<long> ids, List<string> includes, List<OrderBy> orderBys)
+        {
+            var count = SearchCount(ids.OfType<T>().ToList());
+            if (count > 1)
```

Comment:
Three things here:

1. Don't use a global variable.  It doesn't really make sense to change this value on every request - the state means nothing for this.
2. Don't do LastOrDefault(), do Single() - because you're selecting just one record and your count/paging is telling you which record to select.
3. The if logic as you have it is not required - just do the paging no matter what if you have 1 result.  However, for 0 results, just return null.

Soooo I would have:

`
var count = SearchCount(ids.OfType<T>().ToList());
if (count == 0)
{
    return null;
}
var paging = new Paging {Count = 1, Page = count};
return Get(ids, includes, orderBys, paging).Single();
`